### PR TITLE
[gstreamer] build with 1.19.3

### PR DESCRIPTION
**Pull request by luncliff**
Redirect https://github.com/microsoft/vcpkg/pull/22910 to working branch

### CODE REVIEW

**Comment by luncliff** [comment]
I will resolve the conflicts soon

**Comment by luncliff** [comment]
Qt Multimedia's `EGL::EGL` is not what I understand. :(

Feel free to mention me in related issues, @JackBoosY

File: ports/gstreamer/portfile.cmake [hunk 1]
```diff
@@ -1,83 +1,11 @@
-vcpkg_from_github(
```
Comment (JackBoosY) [inline ports/gstreamer/portfile.cmake:1]: Does the new version merge these repos?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/gstreamer/portfile.cmake (hunk 1 above)
Comment (luncliff) [reply ports/gstreamer/portfile.cmake:1]: Yes. They are all merged into `gstreamer` repo's `subprojects` folder. It seems like 1.19.2 was the preparation for the reorganization.

File: ports/gstreamer/portfile.cmake [hunk 2]
```diff
@@ -97,58 +25,64 @@ else()
...
+elseif(VCPKG_TARGET_IS_ANDROID)
+    list(APPEND PLATFORM_OPTIONS
+        -Dgst-plugins-base:gl_platform=egl
+        -Dgst-plugins-base:gl_winsys=android
+    ) 
+endif()
+
 vcpkg_configure_meson(
-    SOURCE_PATH ${GST_BUILD_SOURCE_PATH}
+    SOURCE_PATH ${GST_SOURCE_PATH}
```
Comment (JackBoosY) [inline ports/gstreamer/portfile.cmake]:
```suggestion
    SOURCE_PATH ${SOURCE_PATH}
```

File: ports/gstreamer/portfile.cmake [hunk 3]
```diff
@@ -1,83 +1,11 @@
-vcpkg_from_github(
-    OUT_SOURCE_PATH GST_BUILD_SOURCE_PATH
-    REPO gstreamer/gst-build
-    REF 1.19.2
-    SHA512 d6b8e9fc195a60dfb83fe8a49040c21ca5603e3ada2036d56851e6e61a1cd2653ad45f33e39388bde859dfb4806f4a60d9dbfac5fe41b6d2a8b395c44d4525e3
-    HEAD_REF master
-)
 vcpkg_from_github(
     OUT_SOURCE_PATH GST_SOURCE_PATH
```
Comment (JackBoosY) [inline ports/gstreamer/portfile.cmake]:
```suggestion
    OUT_SOURCE_PATH SOURCE_PATH
```

File: ports/gstreamer/portfile.cmake [hunk 4]
```diff
@@ -252,42 +166,25 @@ if (NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
...
 
-vcpkg_fixup_pkgconfig()
-
-# Remove duplicated GL headers (we already have `opengl-registry`)
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/KHR"
-                    "${CURRENT_PACKAGES_DIR}/include/GL"
-)
-file(RENAME "${CURRENT_PACKAGES_DIR}/lib/gstreamer-1.0/include/gst/gl/gstglconfig.h"
-            "${CURRENT_PACKAGES_DIR}/include/gstreamer-1.0/gst/gl/gstglconfig.h"
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/${pkg_name}/pkgconfig"
```
Comment (JackBoosY) [inline ports/gstreamer/portfile.cmake:169]: Why remove this folder?

File: ports/gstreamer/portfile.cmake [hunk 5]
```diff
@@ -252,42 +166,25 @@ if (NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
...
-    set(SUFFIX ${CMAKE_SHARED_LIBRARY_SUFFIX})
-    file(REMOVE "${CURRENT_PACKAGES_DIR}/debug/lib/${PREFIX}gstreamer-full-1.0${SUFFIX}"
-                "${CURRENT_PACKAGES_DIR}/lib/${PREFIX}gstreamer-full-1.0${SUFFIX}"
-    )
-endif()
-if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
-    file(GLOB DBG_BINS "${CURRENT_PACKAGES_DIR}/debug/lib/gstreamer-1.0/*.dll"
-                       "${CURRENT_PACKAGES_DIR}/debug/lib/gstreamer-1.0/*.pdb"
+elseif(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+    file(GLOB DBG_BINS "${CURRENT_PACKAGES_DIR}/debug/lib/${pkg_name}/*.dll"
```
Comment (JackBoosY) [inline ports/gstreamer/portfile.cmake:179]: I think we should make a patch to fix that.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/gstreamer/portfile.cmake (hunk 4 above)
Comment (luncliff) [reply ports/gstreamer/portfile.cmake:169]: At this point, they are copied from `lib/gstreamer-1.0/pkgconfig` to `lib/pkgconfig`. The folder will be redundant.

File: ports/gstreamer/portfile.cmake (hunk 5 above)
Comment (luncliff) [reply ports/gstreamer/portfile.cmake:179]: I will check if it can be configured in meson.build

Comment (luncliff) [reply ports/gstreamer/portfile.cmake:179]: I couldn't find a way to separate `bin` and `lib` with the project's build option. :(

I studied a little. [`library` has an option `install_dir`](https://mesonbuild.com/Reference-manual_functions.html#arguments36), and it becomes the destination of the library files, DLLs, and PDB.

Comment (JackBoosY) [reply ports/gstreamer/portfile.cmake:179]: I will take a look tomorrow.

Comment (JackBoosY) [reply ports/gstreamer/portfile.cmake:179]: I found the install path is by default, @Neumann-A do you know why?

Comment (Neumann-A) [reply ports/gstreamer/portfile.cmake:179]: No I don't know why gstreamer does this but probably versioned installed dirs are a good default if you don't install versioned libs ?

**Comment by JackBoosY** [comment]
@luncliff Do you have any other changes left?

**Comment by luncliff** [comment]
> @luncliff Do you have any other changes left?

No. I think the changes are enough.

**Pull request merged**